target/linux/realtek/base-files/etc/board.d/02_network

```diff
@@ -8,6 +8,9 @@ ucidef_set_poe() {
 		json_add_string "budget" "$1"
 		json_select_array ports
 			for port in $2; do
+				if [ -n "$3" ] && [ -z "${3##*$port*}" ]; then
+					continue # skip ports passed via $3
```

Comment:
Please fix your globbing here: you're catching e.g. `lan2` on `lan20`, `lan1` on `lan10`, etc.

```txt
[mkennedy@imp ~]$ set 1 "lan1 lan2 lan3 lan4 lan5 lan6 lan7 lan8 lan9" "lan10 lan20"
[mkennedy@imp ~]$ for port in $2; do if [ -n "$3" ] && [ -z "${3##*$port*}" ]; then echo $port continue; fi; done
lan1 continue
lan2 continue
```

Reply:
(I went ahead and retested in an OpenWrt `ash` environment, same result. That said, there might be some other shopts that apply during `board.d/*` that I'm not aware of.)

Reply:
Using `expr match` appears to work on one of my APs running OpenWrt:

```
# set 1 "lan1 lan2 lan3 lan4 lan5 lan6 lan7 lan8 lan9 lan10" "lan10 lan20"
# for port in $2; do if [ -n "$3"  ] && $(expr match "$3" "$port\([[:space:]]\|\$\)"); then echo skip $port; fi; done
skip lan10
```